### PR TITLE
script directory paths and infrastructure templates

### PR DESCRIPTION
**Pull request by bryan-bar**
* Scripts updated to know their own paths
  * results directory will be created under the benchmark directory
  * terraform project will be created under the benchmark directory
  * ref pr: https://github.com/EnterpriseDB/edb-benchmarks/pull/88
* jinja Infrastructure templates added and generated with an ansible playbook
  * Follow-up PR will expose some of these values through environment variables
  * added in first pass
    * `aws-dbt2-biganimal`
    * `aws-dbt2-rds`
    * `aws-edb-postgres-tuner-dbt2`
    * `aws-hammerdb-tproc-c-biganimal`
    * `aws-hammerdb-tproc-c-biganimal-epas`
    * `aws-hammerdb-tproc-c-oracle`
  * some benchmarks such as `aws-pgbench-postgresql` require additional handling since it assumes attached storage
  * others will be updated as needed

### CODE REVIEW

File: aws-dbt2-biganimal/04_push_results/run.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push DBT2 files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-dbt2-biganimal/04_push_results/run.sh:2]: This is missing setting SOURCEDIR.  I recently made that mistake elsewhere. :)

File: aws-dbt2-biganimal/05_unprovision/run.sh [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
-# We need the absolute path of $TERRAFORM_PROJECT_PATH in this script.
-TERRAFORM_PROJECT_PATH="../terraform"
-TERRAFORM_PROJECT_PATH=$(readlink -f "${TERRAFORM_PROJECT_PATH}")
+SOURCEDIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
+TERRAFORM_PROJECT_NAME="terraform"
+TERRAFORM_PROJECT_PATH="${SOURCEDIR}/../${TERRAFORM_PROJECT_NAME}"
 
-cd "${TERRAFORM_PROJECT_PATH}"
+cd ${TERRAFORM_PROJECT_PATH}
```
Comment (mw2q) [inline aws-dbt2-biganimal/05_unprovision/run.sh:7]: Your linter doesn't complain about missing double quotes? :)  I think mine is using shellcheck, if you were wondering...

File: aws-dbt2-rds/04_push_results/run.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push DBT2 files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-dbt2-rds/04_push_results/run.sh:2]: Ditto missing SOURCEDIR

File: aws-edb-postgres-tuner-dbt2/04_push_results/run.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push DBT2 files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-edb-postgres-tuner-dbt2/04_push_results/run.sh:2]: Ditto missing SOURCEDIR

File: aws-hammerdb-tproc-c-biganimal/04_push_results/run.sh [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push HammerDB files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-hammerdb-tproc-c-biganimal/04_push_results/run.sh:2]: ditto missing SOURCEDIR

File: aws-hammerdb-tproc-c-biganimal-epas/04_push_results/run.sh [hunk 6]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push HammerDB files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-hammerdb-tproc-c-biganimal-epas/04_push_results/run.sh:2]: ditto about SOURCEDIR

File: aws-hammerdb-tproc-c-oracle/04_push_results/run.sh [hunk 7]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push HammerDB files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-hammerdb-tproc-c-oracle/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-lsr-3-nodes/04_push_results/run.sh [hunk 8]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push pgbench files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-lsr-3-nodes/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-pgbench-postgresql/05_push_results/run.sh [hunk 9]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eux
-# Push pgbench files to the S3 bucket
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-pgbench-postgresql/05_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-pgd-3-nodes/04_push_results/run.sh [hunk 10]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-pgd-3-nodes/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-psr-3-nodes/04_push_results/run.sh [hunk 11]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-psr-3-nodes/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-tde-pg-upgrade/04_push_results/run.sh [hunk 12]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-tde-pg-upgrade/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-tpcc-pgd/05_push_results/run.sh [hunk 13]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-tpcc-pgd/05_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-tpcc-refdata/05_push_results/run.sh [hunk 14]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-tpcc-refdata/05_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-tpcc-tde/04_push_results/run.sh [hunk 15]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-tpcc-tde/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-tpcc-tde-medium-amd/04_push_results/run.sh [hunk 16]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-tpcc-tde-medium-amd/04_push_results/run.sh:2]: ditto SOURCEDIR

File: aws-tpcc-xdb/04_push_results/run.sh [hunk 17]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -eux
+RESULTS_DIRECTORY="${SOURCEDIR}/../results"
```
Comment (mw2q) [inline aws-tpcc-xdb/04_push_results/run.sh:2]: ditto SOURCEDIR

**Review by mw2q — CHANGES_REQUESTED** [review]
Other than minor shellcheck linting, I think the SOURCEDIR comments are the only things that really need to be fixed.

File: aws-dbt2-biganimal/05_unprovision/run.sh (hunk 2 above)
Comment (bryan-bar) [reply aws-dbt2-biganimal/05_unprovision/run.sh:7]: No, it was not. 
Thank you for the suggestion, I am now using it.

**Review by mw2q — APPROVED** [review]

**Pull request merged**